src/Timers.cpp

```diff
@@ -210,6 +225,218 @@ bool Timers::FindTagInTimerTags(std::string tag, std::string tags)
+
+    if (avoidDuplicateDescription != AUTOTIMER_AVOID_DUPLICATE_DISABLED)
+    {
+      if (searchForDuplicateDescription == AUTOTIMER_CHECK_SEARCH_FOR_DUP_IN_TITLE)
+        autoTimer.deDup = AutoTimer::DeDup::CHECK_TITLE;
+      else if (searchForDuplicateDescription == AUTOTIMER_CHECK_SEARCH_FOR_DUP_IN_TITLE_AND_SHORT_DESC)
+        autoTimer.deDup = AutoTimer::DeDup::CHECK_TITLE_AND_SHORT_DESC;
+      else if (searchForDuplicateDescription == AUTOTIMER_CHECK_SEARCH_FOR_DUP_IN_TITLE_AND_ALL_DESCS)
+        autoTimer.deDup = AutoTimer::DeDup::CHECK_TITLE_AND_ALL_DESCS;
+    }
+
+    if (pNode->QueryStringAttribute("encoding", &encoding) == TIXML_SUCCESS)
+      autoTimer.encoding = encoding;    
+
+    if (pNode->QueryStringAttribute("searchType", &searchType) == TIXML_SUCCESS)
+    {
+      autoTimer.searchType = searchType;
+      if (searchType == AUTOTIMER_SEARCH_TYPE_DESCRIPTION)
+        autoTimer.searchFulltext = true;
+    }
+
+    if (pNode->QueryStringAttribute("searchCase", &searchCase) == TIXML_SUCCESS)
+      autoTimer.searchCase = searchCase;
+
+    TiXmlElement* serviceNode = pNode->FirstChildElement("e2service");
+
+    if (serviceNode)
+    {
+      const TiXmlElement *nextServiceNode = serviceNode->NextSiblingElement("e2service");
+
+      if (!nextServiceNode)
+      {
+        //If we only have one channel
+        if (XMLUtils::GetString(serviceNode, "e2servicereference", strTmp))
+        {
+          autoTimer.iChannelId = vuData.GetChannelNumber(strTmp.c_str());
+          autoTimer.strChannelName = vuData.GetChannels().at(autoTimer.iChannelId-1).strChannelName;  
+        }
+      }
+      else //otherwise set to any channel
+      {
+        autoTimer.iChannelId = PVR_TIMER_ANY_CHANNEL;
+        autoTimer.anyChannel = true; 
+      }
+    } 
+
+    autoTimer.iWeekdays = 0;
+    
+    TiXmlElement* includeNode = pNode->FirstChildElement("include");
+
+    if (includeNode)
+    {
+      for (; includeNode != NULL; includeNode = includeNode->NextSiblingElement("include"))
+      {
+        std::string includeVal = includeNode->GetText();
+
+        std::string where;
+        if (includeNode->QueryStringAttribute("where", &where) == TIXML_SUCCESS)
+        {
+          if (where == "dayofweek")
+          {
+            autoTimer.iWeekdays |= (1 << atoi(includeVal.c_str()));
+          }
+        }
+      }
+    }
+
+    if (autoTimer.iWeekdays != 0)
+    {
+      std::time_t t = std::time(nullptr);
+      std::tm timeinfo = *std::localtime(&t);
+      timeinfo.tm_sec = 0;
+      autoTimer.startTime = 0;
+      if (from != "")
+      {
+        ParseTime(from, timeinfo);
+        autoTimer.startTime = std::mktime(&timeinfo);
+      }
+
+      timeinfo = *std::localtime(&t);
+      timeinfo.tm_sec = 0;
+      autoTimer.endTime = 0;
+      if (to != "")
+      {
+        ParseTime(to, timeinfo);
+        autoTimer.endTime = std::mktime(&timeinfo);
+      }
+    }
+    else
+    {
+      for (int i = 0; i < 7; i++)
```

Comment:
`7`? Where does this magic number come from. You should define a `static const int MEANINGFUL_NAME = 7;`either here or if that magic 7 is used elsewhere in the header of `Timers`class.